blockindex/indexer.go

```diff
@@ -47,7 +47,7 @@ type (
 		Stop(context.Context) error
 		Commit() error
 		PutBlock(*block.Block) error
-		DeleteBlock(*block.Block) error
+		DeleteTipBlock(*block.Block) error
```

Comment:
and Indexer becomes:
Indexer interface {
    IndexPutter;
    all remaining API
}